File: matiasmasso/Mat.NET/DTO/DTOCustomerPlatform.cs
namespace DTO
{
    using System;
    using System.Collections.Generic;

    public class DTOCustomerPlatform : DTOContact
    {
        public DTOContact Customer { get; set; }
        public List<DTOPlatformDestination> Destinations { get; set; }
        public List<DTODelivery> Deliveries { get; set; }
        public DTOAmt BaseImponible { get; set; }

        public DTOCustomerPlatform() : base()
        {
        }

        public DTOCustomerPlatform(Guid oGuid) : base(oGuid)
        {
            Destinations = new List<DTOPlatformDestination>();
        }

        public static DTOCustomerPlatform FromContact(DTOContact oContact)
        {
            DTOCustomerPlatform retval = null;
            if (oContact == null)
                retval = new DTOCustomerPlatform();
            else if (oContact is DTOCustomerPlatform)
                retval = (DTOCustomerPlatform)oContact;
            else
            {
                retval = new DTOCustomerPlatform(oContact.Guid);
                {
                    var withBlock = retval;
                    withBlock.Emp = oContact.Emp;
                    withBlock.Nom = oContact.Nom;
                    withBlock.NomComercial = oContact.NomComercial;
                    withBlock.SearchKey = oContact.SearchKey;
                    withBlock.FullNom = oContact.FullNom;
                    withBlock.Nifs = oContact.Nifs;
                    withBlock.Address = oContact.Address;
                    withBlock.ContactClass = oContact.ContactClass;
                    withBlock.Lang = oContact.Lang;
                    withBlock.Rol = oContact.Rol;
                    withBlock.NomAnterior = oContact.NomAnterior;
                    withBlock.NomNou = oContact.NomNou;
                    withBlock.Website = oContact.Website;
                    withBlock.Logo = oContact.Logo;
                    withBlock.GLN = oContact.GLN;
                    withBlock.Telefon = oContact.Telefon;
                    withBlock.Tels = oContact.Tels;
                    withBlock.ContactPersons = oContact.ContactPersons;
                    withBlock.Obsoleto = oContact.Obsoleto;
                }
            }
            return retval;
        }
    }


    public class DTOPlatformDestination : DTOContact
    {
        public DTOCustomerPlatform Platform { get; set; }


        public DTOPlatformDestination() : base()
        {
        }

        public DTOPlatformDestination(Guid oGuid) : base(oGuid)
        {
        }

        public static DTOPlatformDestination FromContact(DTOContact oContact)
        {
            DTOPlatformDestination retval = new DTOPlatformDestination(oContact.Guid);
            retval.Emp = oContact.Emp;
            retval.FullNom = oContact.FullNom;
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOCustomerPlatform.vb
Public Class DTOCustomerPlatform
    Inherits DTOContact

    Public Property Customer As DTOContact
    Public Property Destinations As List(Of DTOPlatformDestination)
    Public Property Deliveries As List(Of DTODelivery)
    Public Property BaseImponible As DTOAmt

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
        _Destinations = New List(Of DTOPlatformDestination)
    End Sub

    Shared Function FromContact(oContact As DTOContact) As DTOCustomerPlatform
        Dim retval As DTOCustomerPlatform = Nothing
        If oContact Is Nothing Then
            retval = New DTOCustomerPlatform
        ElseIf TypeOf oContact Is DTOCustomerPlatform Then
            retval = oContact
        Else
            retval = New DTOCustomerPlatform(oContact.Guid)
            With retval
                .Emp = oContact.Emp
                .Nom = oContact.Nom
                .NomComercial = oContact.NomComercial
                .SearchKey = oContact.SearchKey
                .FullNom = oContact.FullNom
                .Nif = oContact.Nif
                .Nif2 = oContact.Nif2
                .Address = oContact.Address
                .ContactClass = oContact.ContactClass
                .Lang = oContact.Lang
                .Rol = oContact.Rol
                .NomAnterior = oContact.NomAnterior
                .NomNou = oContact.NomNou
                .Website = oContact.Website
                .DisplayWebsite = oContact.DisplayWebsite
                .Botiga = oContact.Botiga
                .Logo = oContact.Logo
                .GLN = oContact.GLN
                .Telefon = oContact.Telefon
                .Tels = oContact.Tels
                .ContactPersons = oContact.ContactPersons
                .Obsoleto = oContact.Obsoleto
            End With
        End If
        Return retval
    End Function
End Class


Public Class DTOPlatformDestination
    Inherits DTOContact

    Public Property Platform As DTOCustomerPlatform


    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function FromContact(oContact As DTOContact) As DTOPlatformDestination
        Dim retval As New DTOPlatformDestination(oContact.Guid)
        retval.Emp = oContact.Emp
        retval.FullNom = oContact.FullNom
        Return retval
    End Function
End Class



